src/js/controller/controller.js

```diff
@@ -1028,11 +1028,14 @@ Object.assign(Controller.prototype, {
         return this._model.getMute();
     },
     triggerError(evt) {
-        this._model.set('errorEvent', evt);
-        this._model.set('state', STATE_ERROR);
-        this._model.once('change:state', function() {
-            this._model.set('errorEvent', undefined);
-        }, this);
+        const model = this._model;
+        evt.message = model.get('localization').errors[evt.key];
+        delete evt.key;
```

Comment:
`delete` causes a deoptimization in v8. I think we should set it to null if acceptable; or create a new object

Reply:
https://github.com/petkaantonov/bluebird/wiki/Optimization-killers

>An object will go into hash table mode for example when you add too many properties dynamically (outside constructor), delete properties, use properties that cannot be valid identifiers and so on. 

>Objects that are in "hash table mode" (aka "normalized objects", "dictionary mode" - objects who have a hash table as a backing data structure) are not simple enumerables**